init_features.go

```diff
@@ -159,68 +184,54 @@ func initFeatures(
 			// actually a pointer to a Feature struct. See BASE CASE
 			// section below.
 
-			if targetFqn != "" {
-				// The target fqn here could be
-				//    - "some_other_feature_in_this_feature_class"
-				//    - "this_windowed_feature__600__"
-				//    - "other_windowed_feature__600__"
-				// So let's extract "this_windowed_feature" from the target fqn
-				// and compare it with the current feature's name.
-				pattern, err := regexp.Compile(fmt.Sprintf(`^%s__\d+__$`, resolvedName))
-				if err != nil {
-					return nil, errors.Wrap(err, "error compiling regex to match windowed feature names")
-				}
-				if !pattern.Match([]byte(getFqnRoot(targetFqn))) {
-					// Not any bucket feature in this windowed feature class
-					continue
-				}
-			}
-
 			mapValueType := f.Type().Elem()
 			if mapValueType.Kind() != reflect.Pointer {
-				return nil, fmt.Errorf(
+				return errors.Newf(
 					"the map type for Windowed features should a pointer"+
 						" as its value type, but found %s instead",
 					mapValueType.Kind(),
 				)
 			}
 
-			if targetFqn == "" {
-				// Initializing all features, always
-				// make a new map.
-				f.Set(reflect.MakeMap(f.Type()))
-				windows := fm.Meta.Tag.Get("windows")
-				for _, tag := range strings.Split(windows, ",") {
-					seconds, parseErr := internal.ParseBucketDuration(tag)
-					if parseErr != nil {
-						return nil, fmt.Errorf("error parsing bucket duration: %s", parseErr)
+			windows := fm.Meta.Tag.Get("windows")
+			for _, tag := range strings.Split(windows, ",") {
+				seconds, err := internal.ParseBucketDuration(tag)
+				if err != nil {
+					return errors.Wrap(err, "error parsing bucket duration: %s")
+				}
+				updatedResolvedName := fmt.Sprintf("%s__%d__", resolvedName, seconds)
+				bucketFqn := fmt.Sprintf("%s.%s", cumulativeFqn, updatedResolvedName)
+				if isScoped {
+					if _, bucketInScope := scope.children[updatedResolvedName]; !bucketInScope {
+						continue
 					}
-					windowFqn := updatedFqn + fmt.Sprintf("__%d__", seconds)
-					if targetFqn == "" {
-						feature := Feature{Fqn: windowFqn}
-						ptrInDisguiseToFeature := reflect.NewAt(mapValueType.Elem(), reflect.ValueOf(&feature).UnsafePointer())
-						f.SetMapIndex(reflect.ValueOf(tag), ptrInDisguiseToFeature)
+					// Make map only if one of the bucket features need to be set
+					// If no bucket features need to be set, map is nil.
+					if f.IsNil() {
+						f.Set(reflect.MakeMap(f.Type()))
 					}
+					fi.fieldsMap[bucketFqn] = append(fi.fieldsMap[bucketFqn], f)
+				} else {
+					if f.IsNil() {
+						f.Set(reflect.MakeMap(f.Type()))
+					}
+					feature := Feature{Fqn: bucketFqn}
+					ptrInDisguiseToFeature := reflect.NewAt(mapValueType.Elem(), reflect.ValueOf(&feature).UnsafePointer())
```

Comment:
This doesn't seem like a safe usage of `UnsafePointer` - I think this should just be

```go
reflect.ValueOf(&feature)
```

Reply:
Discussed offline - this seems wrong but isn't new to this PR so it's not a blocker

Reply:
ticketed here https://linear.app/chalk-ai/issue/CHA-3991/re-evaluate-chalk-go-deser-implementation-via-unsafepointer